lib/sequent/core/helpers/argument_serializer.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+module Sequent
+  module Core
+    module Helpers
+      class ArgumentSerializer
+        class << self
+          def serialize_value(value)
```

Comment:
This class confused me a bit since it has a generic name but is specific to Matchers. Perhaps move this to a Matcher helper or anything?